### PR TITLE
[BugFix] Reset current database after using catalog

### LINKED ISSUE #18862: [Bug] Using catalog does not reset current database
### Steps to reproduce the behavior (Required)
1. `USE catalogA.db;`
2. `USE "CATALOG catalogB";` // `catalogB` does not have a database named `db`
`ERROR 1064 (HY000): Unknown database 'db'`  

### Expected behavior (Required)
Use catalog shall reset current datatbase.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by letian-jiang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #18862

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by letian-jiang** [comment]
run starrocks_be_unittest

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_be_unittest

**Comment by letian-jiang** [comment]
run starrocks_fe_unittest

**Comment by letian-jiang** [comment]
run starrocks_fe_unittest

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Review by Smith-Cruise — COMMENTED** [review]
Can you add a FE ut for it?

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ private void handleInitDb() {
             String[] parts = identifier.trim().split("\\s+");
             if (parts.length == 2) {
                 if (parts[0].equalsIgnoreCase("catalog")) {
-                    ctx.getGlobalStateMgr().changeCatalog(ctx, parts[1]);
+                    ctx.getGlobalStateMgr().changeCatalogDb(ctx, parts[1] + ".");
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java:110]: Although the code can be reused, but the readability is weaker. suggest another way

Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java]: Have restored the `changeCatalog` function.

**Comment by letian-jiang** [comment]
run starrocks_fe_unittest

**Comment by letian-jiang** [comment]
run starrocks_admit_test

File: fe/fe-core/src/test/java/com/starrocks/catalog/ChangeCatalogDBTest.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            GlobalStateMgr.getCurrentState().changeCatalogDb(ctx, "default_catalog.nonexistent_db");
+        });
+
+        Assertions.assertThrows(DdlException.class, () -> {
+            GlobalStateMgr.getCurrentState().changeCatalogDb(ctx, "hive_catalog.nonexistent_db");
+        });
+    }
+
+    @AfterAll
+    public static void teardown() {}
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/catalog/ChangeCatalogDBTest.java:137]: delete it

Comment (letian-jiang) [reply fe/fe-core/src/test/java/com/starrocks/catalog/ChangeCatalogDBTest.java]: done

**Review by stephen-shelby — DISMISSED** [review]

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_fe_unittest

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|8|10|80.00%|[3237, 3250]|

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by Smith-Cruise** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Pull request merged**